web/source/kmwdom.ts

```diff
@@ -720,15 +676,20 @@ class DOMManager {
    * @return      {Array<Element>}        A list of potentially-editable controls.  Further filtering [as with isKMWInput() and
    *                                      isKMWDisabled()] is required.
    */
-  _GetDocumentEditables(Pelem: HTMLElement|Document) {
+  _GetDocumentEditables(Pelem: HTMLElement|Document): (HTMLElement|Document)[] {
```

Comment:
I think we could potentially make this clearer by defining something like: 

```
type DocumentEditableElement = HTMLInputElement | HTMLTextAreaElement | HTMLIFrameElement | Document;

_GetDocumentEditables(Pelem: DocumentEditableElement): DocumentEditableElement[] {
}
```

Reply:
I'm not sure if we need to go this deep for this PR, but it would clarify the inputs and outputs to define union types for this (ab)use of typing for the whole project; later we'll then be able to see the patterns of use.

Reply:
Yeah, I only just discovered the ability to use that pattern.  The thing is, though, we also return contentEditable elements of whatever type:  divs, spans, etc.

My preference would be to create a wrapper for each type - input, textarea, iframe, and contenteditable - that handles the DOM aspects for the particularities of each.  That'd be a PR after fixing up the OSK, though, which is way down the road.